gobblin-iceberg/src/main/java/org/apache/gobblin/iceberg/publisher/GobblinMCEPublisher.java

```diff
@@ -115,9 +115,11 @@ public void publishData(Collection<? extends WorkUnitState> states) throws IOExc
   }
 
   protected Map<String, String> getPartitionOffsetRange(String offsetKey) {
-    return state.getPropAsList(offsetKey)
```

Comment:
A common idiomatic way of handling this nullable case is
```
Optional.ofNullable(...)
.orElseGet(Collections::emptyList)
.stream()
```

Reply:
Ah, in that case "orElse(null)" is what you'd want.

But either way, seems like this is a headache to handle since everything can't handle null 😆 . Java stream / optional API can only help so much